src/main/java/no/fint/betaling/service/FileService.java

```diff
@@ -0,0 +1,178 @@
+        return s -> Optional.ofNullable(s.getOrganisasjonsnummer())
+                .map(Identifikator::getIdentifikatorverdi)
+                .map(schoolId::equals)
+                .orElse(false);
+    }
+
+    private List<Customer> getCustomersForGroup(List<Link> studentRelationLinks) {
+        if (studentRelationLinks == null) {
+            return Collections.emptyList();
+        }
+        Map<Link, ElevforholdResource> studentRelations = fileRepository.getStudentRelations();
+        Map<Link, PersonResource> students = fileRepository.getStudents();
+
+        return studentRelationLinks.stream()
+                .map(studentRelations::get)
+                .filter(Objects::nonNull)
+                .map(this::getStudentLink)
+                .filter(Objects::nonNull)
+                .map(students::get)
+                .filter(Objects::nonNull)
+                .map(CustomerFactory::toCustomer)
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    private Link getStudentLink(ElevforholdResource resource) {
+        return resource.getElev().stream().findFirst().orElse(null);
+    }
+
+    public CustomerFileGroup getCustomersFromFile(String schoolId, byte[] file) throws UnableToReadFileException, NoVISIDColumnException {
+        InputStream targetStream = new ByteArrayInputStream(file);
+        Workbook wb;
+        try {
+            wb = WorkbookFactory.create(targetStream);
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
+            throw new UnableToReadFileException(ex.getMessage());
+        }
+        Sheet sheet = wb.getSheetAt(0);
+        return extractCustomerFileGroupFromFile(sheet, schoolId);
+    }
+
+    private CustomerFileGroup extractCustomerFileGroupFromFile(Sheet sheet, String schoolId) throws NoVISIDColumnException {
+        CustomerFileGroup customerFileGroup = new CustomerFileGroup();
+        CustomerGroup customerGroup = new CustomerGroup();
+        ArrayList<Customer> customers = new ArrayList<>();
+        ArrayList<String> notFoundCustomers = new ArrayList<>();
+
+        Cell visIdCell = getVisIdCell(sheet);
+
+        IntStream
+                .range(visIdCell.getRowIndex()+1, sheet.getPhysicalNumberOfRows())
+                .mapToObj(sheet::getRow)
+                .filter(Objects::nonNull)
+                .map(row -> row.getCell(visIdCell.getColumnIndex()))
+                .filter(Objects::nonNull)
+                .map(cell1 -> String.format("%.0f", cell1.getNumericCellValue()))
+                .forEach(s ->
+                    getCustomerNumberByVisId(s, schoolId)
+                            .map(customers::add)
+                            .orElseGet(() -> notFoundCustomers.add(s))
+                );
+
+        customerGroup.setCustomers(customers);
+        customerFileGroup.setFoundCustomers(customerGroup);
+        customerFileGroup.setNotFoundCustomers(notFoundCustomers);
+        return customerFileGroup;
```

Comment:
Since `CustomerFileGroup` has `@AllArgsConstructor`, you could `return new CustomerFileGroup(customerGroup, notFoundCustomers);` instead.

Reply:
Updated